bootstrap-vcpkg.bat failing

**Describe the bug**
`bootstrap-vcpkg.bat` faile with error:

````
..\src\vcpkg\versiondeserializers.cpp(112): error C2440: 'initializing': cannot convert from 'initializer list' to 'vcp
kg::SchemedVersion' [C:\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
  ..\src\vcpkg\versiondeserializers.cpp(112): note: No constructor could take the source type, or constructor overload
  resolution was ambiguous
````

**Environment**
- OS: Windows 10 Pro, x64 
- Compiler: VS 2015, Update 3

**To Reproduce**
Steps to reproduce the behavior:
1. Just run `bootstrap-vckpg.bat`
2. Fails with the above error

**Expected behavior**
It should.. build.

**Failure logs** 
````
  versiondeserializers.cpp
..\src\vcpkg\versiondeserializers.cpp(112): error C2440: 'initializing': cannot convert from 'initializer list' to 'vcp
kg::SchemedVersion' [C:\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
  ..\src\vcpkg\versiondeserializers.cpp(112): note: No constructor could take the source type, or constructor overload
  resolution was ambiguous
C:\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed. Please ensure you have installed Visual Studio with the
Desktop C++ workload and the Windows SDK for Desktop C++.
At line:1 char:4
+ & {& 'C:\vcpkg\scripts\bootstrap.ps1' }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,bootstrap.ps1

ScriptHalted
At C:\vcpkg\scripts\bootstrap.ps1:411 char:5
+     throw
+     ~~~~~
    + CategoryInfo          : OperationStopped: (:) [], RuntimeException
    + FullyQualifiedErrorId : ScriptHalted
````

**Additional context**
None.

I have the same issue (VS2015 update3, Win10 20H2) using the latest sources (c2329ab).

```
  visualstudio.cpp
..\src\vcpkg\versiondeserializers.cpp(112): error C2440: 'initializing': cannot convert from 'initializer list' to 'vcp
kg::SchemedVersion' [D:\projects\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
  ..\src\vcpkg\versiondeserializers.cpp(112): note: No constructor could take the source type, or constructor overload
  resolution was ambiguous
  vcpkg.cpp
D:\projects\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed. Please ensure you have installed Visual Studio
with the Desktop C++ workload and the Windows SDK for Desktop C++.
In Zeile:1 Zeichen:4
+ & {& 'd:\projects\vcpkg\scripts\bootstrap.ps1' -disableMetrics}
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,bootstrap.ps1

ScriptHalted
In D:\projects\vcpkg\scripts\bootstrap.ps1:411 Zeichen:5
+     throw
+     ~~~~~
    + CategoryInfo          : OperationStopped: (:) [], RuntimeException
    + FullyQualifiedErrorId : ScriptHalted
```

Likewise. identical issue, though my output lists vcpkg.cpp just before the first failure. This is from a pull from origin/master not 5 minute ago, so I have the "latest source". 

```
  visualstudio.cpp
  vcpkg.cpp
..\src\vcpkg\versiondeserializers.cpp(112): error C2440: 'initializing': cannot convert from 'initializer list' to 'vcp
kg::SchemedVersion' [C:\Projects\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
  ..\src\vcpkg\versiondeserializers.cpp(112): note: No constructor could take the source type, or constructor overload
  resolution was ambiguous
C:\Projects\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed. Please ensure you have installed Visual Studio
with the Desktop C++ workload and the Windows SDK for Desktop C++.
At line:1 char:4
+ & {& 'C:\Projects\vcpkg\scripts\bootstrap.ps1' }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,bootstrap.ps1

ScriptHalted
At C:\Projects\vcpkg\scripts\bootstrap.ps1:411 char:5
+     throw
+     ~~~~~
    + CategoryInfo          : OperationStopped: (:) [], RuntimeException
    + FullyQualifiedErrorId : ScriptHalted

```

On a hunch, I picked the 2020.07 tag and gave it another go. No issues:

```
  versiont.cpp
  visualstudio.cpp
  vcpkglib.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkglib.lib
  vcpkg.cpp
  Generating code
  All 25384 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkg.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkg.exe
  vcpkg.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkg.pdb (Full PDB)
  vcpkgmetricsuploader.cpp
  Generating code
  All 11117 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkgmetricsuploader.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkgmetricsuploader.exe
  vcpkgmetricsuploader.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkgmetricsuploader.pdb (Full PDB)

Building vcpkg.exe... done.

Telemetry
---------
vcpkg collects usage data in order to help us improve your experience.
The data collected by Microsoft is anonymous.
You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
passing --disable-metrics to vcpkg on the command line,
or by setting the VCPKG_DISABLE_METRICS environment variable.

Read more about vcpkg telemetry at docs/about/privacy.md
```

So.. whatever broke, it was between those commits.

> Likewise. identical issue, though my output lists vcpkg.cpp just before the first failure. This is from a pull from origin/master not 5 minute ago, so I have the "latest source".
> 
> ```
>   visualstudio.cpp
>   vcpkg.cpp
> ..\src\vcpkg\versiondeserializers.cpp(112): error C2440: 'initializing': cannot convert from 'initializer list' to 'vcp
> kg::SchemedVersion' [C:\Projects\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
>   ..\src\vcpkg\versiondeserializers.cpp(112): note: No constructor could take the source type, or constructor overload
>   resolution was ambiguous
> C:\Projects\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed. Please ensure you have installed Visual Studio
> with the Desktop C++ workload and the Windows SDK for Desktop C++.
> At line:1 char:4
> + & {& 'C:\Projects\vcpkg\scripts\bootstrap.ps1' }
> +    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
>     + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,bootstrap.ps1
> 
> ScriptHalted
> At C:\Projects\vcpkg\scripts\bootstrap.ps1:411 char:5
> +     throw
> +     ~~~~~
>     + CategoryInfo          : OperationStopped: (:) [], RuntimeException
>     + FullyQualifiedErrorId : ScriptHalted
> ```
> 
> On a hunch, I picked the 2020.07 tag and gave it another go. No issues:
> 
> ```
>   versiont.cpp
>   visualstudio.cpp
>   vcpkglib.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkglib.lib
>   vcpkg.cpp
>   Generating code
>   All 25384 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
>   Finished generating code
>   vcpkg.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkg.exe
>   vcpkg.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkg.pdb (Full PDB)
>   vcpkgmetricsuploader.cpp
>   Generating code
>   All 11117 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
>   Finished generating code
>   vcpkgmetricsuploader.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkgmetricsuploader.exe
>   vcpkgmetricsuploader.vcxproj -> C:\Projects\vcpkg\toolsrc\msbuild.x86.release\vcpkgmetricsuploader.pdb (Full PDB)
> 
> Building vcpkg.exe... done.
> 
> Telemetry
> ---------
> vcpkg collects usage data in order to help us improve your experience.
> The data collected by Microsoft is anonymous.
> You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
> passing --disable-metrics to vcpkg on the command line,
> or by setting the VCPKG_DISABLE_METRICS environment variable.
> 
> Read more about vcpkg telemetry at docs/about/privacy.md
> ```
> 
> So.. whatever broke, it was between those commits.

Same issue, resolved by using 2020.07 instead of master.

--- a/toolsrc/include/vcpkg/versiondeserializers.h
+++ b/toolsrc/include/vcpkg/versiondeserializers.h
@@ -22,6 +22,9 @@ namespace vcpkg
 
     struct SchemedVersion
     {
+        SchemedVersion() = default;
+        SchemedVersion(Versions::Scheme scheme_, VersionT versiont_) : scheme(scheme_), versiont(versiont_) {}
+
         Versions::Scheme scheme = Versions::Scheme::String;
         VersionT versiont;
     };

--- a/toolsrc/include/vcpkg/versiondeserializers.h
+++ b/toolsrc/include/vcpkg/versiondeserializers.h
@@ -23,7 +23,9 @@ namespace vcpkg
     struct SchemedVersion
     {
         SchemedVersion() = default;
-        SchemedVersion(Versions::Scheme scheme_, VersionT versiont_) : scheme(scheme_), versiont(std::move(versiont_)) {}
+        SchemedVersion(Versions::Scheme scheme_, VersionT versiont_) : scheme(scheme_), versiont(std::move(versiont_))
+        {
+        }
 
         Versions::Scheme scheme = Versions::Scheme::String;
         VersionT versiont;

--- a/toolsrc/include/vcpkg/versiondeserializers.h
+++ b/toolsrc/include/vcpkg/versiondeserializers.h
@@ -23,7 +23,7 @@ namespace vcpkg
     struct SchemedVersion
     {
         SchemedVersion() = default;
-        SchemedVersion(Versions::Scheme scheme_, VersionT versiont_) : scheme(scheme_), versiont(versiont_) {}
+        SchemedVersion(Versions::Scheme scheme_, VersionT versiont_) : scheme(scheme_), versiont(std::move(versiont_)) {}
 
         Versions::Scheme scheme = Versions::Scheme::String;
         VersionT versiont;

--- a/toolsrc/src/vcpkg/versiondeserializers.cpp
+++ b/toolsrc/src/vcpkg/versiondeserializers.cpp
@@ -109,7 +109,7 @@ namespace vcpkg
                 Checks::unreachable(VCPKG_LINE_INFO);
         }
 
-        return SchemedVersion{version_scheme, {version, port_version}};
+        return SchemedVersion(version_scheme, VersionT{version, port_version});
     }
 
     View<StringView> schemed_deserializer_fields()

